framework/include/reporters/SolutionInvalidityReporter.h

```diff
@@ -0,0 +1,34 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "GeneralReporter.h"
+
+class SolutionInvalidity;
+
+/**
+ * Reports the summary table of solution invalid warnings
+ */
+class SolutionInvalidityReporter : public GeneralReporter
+{
+public:
+  static InputParameters validParams();
+
+  SolutionInvalidityReporter(const InputParameters & parameters);
+
+  void initialize() override {}
+  void finalize() override {}
+  void execute() override {}
+
+private:
+  const SolutionInvalidity * const & _solution_invalidity;
```

Comment:
reference to a pointer? Can we get away with just a reference? Is the reporter created before or after the SolutionInvalidity?

Reply:
I think the report is created first. If I use only a reference, there will be a compile error like `no matching function for call to object of type 'const nlohmann::detail::to_json_fn'. @loganharbour suggested using a "reference to a pointer".

Reply:
This is how reporters work. 

Now that I look at it - I don't think we even need to capture this reference. Can you try removing this variable and then just moving the declare call in the source file out of the initializer list (no variable to assign it to)?

Reply:
This comment has been addressed.